### PR TITLE
improvement: track `supertree`, and add `Zipper.all_the_way_up/1`

### PR DESCRIPTION
**Pull request by zachdaniel**
I don't really think that `all_the_way_up/1` is a great name, and I had to resist the urge to call it `Zipper.tippy_top/1` 😂

Open to suggestions.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+elixir 1.16.3
```
Comment (doorgan) [inline .tool-versions:1]: This was accidentally commited?

File: lib/sourceror/zipper.ex [hunk 2]
```diff
@@ -81,12 +84,22 @@ defmodule Sourceror.Zipper do
...
 
+  @doc """
+  Walks the `zipper` all the way up, breaking out of any subtrees and returns the top-most `zipper`.
+  """
+  @spec all_the_way_up(t) :: t
+  def all_the_way_up(%Z{supertree: supertree} = zipper) when not is_nil(supertree) do
+    all_the_way_up(into(top(zipper), supertree))
+  end
+
+  def all_the_way_up(zipper), do: top(zipper)
```
Comment (doorgan) [inline lib/sourceror/zipper.ex:101]: Maybe we can call this `topmost`? It feels a bit more consistent with the names `rightmost` and `leftmost`

**Comment by doorgan** [comment]
Should we also include `topmost_root` mimic `root` but that can be called from a subtree too?

File: .tool-versions (hunk 1 above)
Comment (zachdaniel) [reply .tool-versions:1]: yes :) When testing, there appear to be errors on 1.17 so I did this to pin it to 1.16.3 🥶 could very easily have been a false positive or something dumb I was doing.

File: lib/sourceror/zipper.ex (hunk 2 above)
Comment (zachdaniel) [reply lib/sourceror/zipper.ex:101]: I like `topmost` 👍

**Comment by zachdaniel** [comment]
Happy to add `topmost_root` as well :)

**Pull request merged**